### ISSUE TITLE
Unexpected log-out, history lost

### ISSUE THREAD

**Issue by Lorak-mmk** [wireapp/wire-desktop#2833]
Wire version: 3.9.2895
Operating system: Linux manjaro 5.1.16-1-MANJARO #1 SMP PREEMPT Thu Jul 4 20:32:22 UTC 2019 x86_64 GNU/Linux
Which antivirus software do you have installed: None

What steps will reproduce the problem?
Wire hanged for no reason, so I restarted it (tray icon -> close). After opening app, it asked me to login (at which point I made a copy of ~/.config/Wire folder). After logging in, there is no conversation history, and it's recognized as new device.

What can I do to restore conversation history? ~/.config/Wire folder's size is 320M, so the data is probably there, question is how to get it back. Crashes, slow working and other problems are tolerable, especially in beta version, but total loss of history and log-out for totally no reason is extremely bad, I hope there is some way to fix it.

**Comment by selurvedu** [comment]
Same here on Manjaro. Fortunately, I had all my history on another device.
This may be a distro-specific problem, since it appeared as soon as I upgraded to [wire-desktop 3.9.2895-4](https://git.archlinux.org/svntogit/community.git/commit/trunk?h=packages/wire-desktop&id=bac03ebfdcb73fa179a1d01ee2a65a13e657a5bc).

CC @maximbaz

**Comment by maximbaz** [comment]
Thanks for the ping.

Is it possible that Manjaro got `electron` package upgraded to v5 before it got `wire-desktop` upgraded to `3.9.2895-4`? If these two packages weren't upgraded on your system _at the same time_, you'd experience this loss of history.

In the Arch repos we are trying to synchronize "breaking" releases, so people would get both packages upgraded at the same time, but I'm not sure Manjaro is upgrading packages in the same fashion...?

To give you some technical details, once `electron` package got upgraded to v5, next time Wire would start against electron v5 and "kinda work", but it would be a little broken (like some shortcuts would not work, you wouldn't be able to open preferences window, it would suddenly hang), but once `wire-desktop` gets upgraded to `-4` and locks itself to `electron4`, the behavior fixes but... it seems electron v5 corrupts data in `~/.config/Wire` and so `electron4` is unable to read it properly, and thus logs you our and you lose your history.

Sadly I don't know if it's possible to recover the data in `~/.config/Wire` and get your history back 😞

**Comment by Lorak-mmk** [comment]
Yes, it seems that in update released on 2019.07.03 electron got updated from 4.1.4-1 to 5.0.6-1, without updating wire, in 2019.07.09 update wire got updated from 3.9.2895-3 to 3.9.2895-4, no electron update, and electron4 update from 4.2.5-1 to 4.2.6-1.
I think, that even if dependencies are incorrect, it shouldn't result in log-out in history loss, that should never happen. Is it possible to implement some check and instead of deleting everything, giving some info about wrong dependencies and quitting? It seems like much more sane behaviour.

**Comment by selurvedu** [comment]
> To give you some technical details, once electron package got upgraded to v5, next time Wire would start against electron v5 and "kinda work", but it would be a little broken (like some shortcuts would not work, you wouldn't be able to open preferences window, it would suddenly hang), but once wire-desktop gets upgraded to -4 and locks itself to electron4, the behavior fixes but... it seems electron v5 corrupts data in ~/.config/Wire and so electron4 is unable to read it properly, and thus logs you our and you lose your history.

This is exactly what happened. Thanks for clarifying.

**Comment by maximbaz** [comment]
OK, thanks for confirming 👍 

> I think, that even if dependencies are incorrect, it shouldn't result in log-out in history loss, that should never happen. Is it possible to implement some check and instead of deleting everything, giving some info about wrong dependencies and quitting? It seems like much more sane behaviour.

I think this is a good idea, but I'm not sure how much control Wire devs actually have against this specific case - if Electron is doing something to files in `~/.config/Wire` _before_ Wire code is even getting executed (perhaps it is migrating some system files to a new format), then by the time Wire code is getting executed, it's already too late and the files are not compatible with previous Electron versions.

What would also make such issues a lot less painful is if backup-restore functionality was cross-platform, not many people have Wire installed on two desktops, but almost everyone has Wire on desktop and on a mobile device.

**Comment by selurvedu** [comment]
> What would also make such issues a lot less painful is if backup-restore functionality was cross-platform, not many people have Wire installed on two desktops, but almost everyone has Wire on desktop and on a mobile device.

+1

**Comment by fermulator** [comment]
similar to https://github.com/wireapp/wire-desktop/issues/2114 ?
